src/services/repos/useRepos.tsx

```diff
@@ -0,0 +1,271 @@
+import { useInfiniteQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import { RepoConfig } from 'services/repo/useRepoConfig'
+import Api from 'shared/api'
+import { mapEdges } from 'shared/utils/graphql'
+
+import { orderingOptions } from './config'
+
+const repositoryFragment = `
+  fragment RepoForList on Repository {
+    name
+    active
+    activated
+    private
+    coverage
+    updatedAt
+    latestCommitAt
+    lines
+    author {
+      username
+    }
+    repositoryConfig {
+      indicationRange {
+        upperRange
+        lowerRange
+      }
+    }
+  }
+`
+
+type RepoQueryVariables = {
+  filters: {
+    activated?: boolean
+    term?: string
+    repoNames?: string[]
+    isPublic?: boolean | null
+  }
+  ordering?: string
+  direction?: string
+  first: number
+}
+
+interface FetchMyReposArgs {
+  provider: string
+  variables: RepoQueryVariables
+  after: string
+  signal?: AbortSignal
+}
+
+const RepositorySchema = z
+  .object({
+    name: z.string(),
+    active: z.boolean(),
+    activated: z.boolean().nullable(),
+    private: z.boolean(),
+    coverage: z.number().nullish(),
+    latestCommitAt: z.string().nullable(),
+    lines: z.number().nullable(),
+    author: z.object({
+      username: z.string().nullable(),
+    }),
+    repositoryConfig: RepoConfig,
+    updatedAt: z.string().nullable(),
+  })
+  .nullable()
+
+export type RepositoryResult = z.infer<typeof RepositorySchema>
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      repositories: z
+        .object({
+          edges: z.array(
+            z.object({
+              node: RepositorySchema,
+            })
+          ),
+          pageInfo: z.object({
+            hasNextPage: z.boolean(),
+            endCursor: z.string().nullable(),
+          }),
+        })
+        .nullable(),
+    })
+    .nullable(),
+})
+
+const RequestSchemaMyRepos = z.object({
+  me: z
+    .object({
+      viewableRepositories: z
+        .object({
+          edges: z.array(
+            z.object({
+              node: RepositorySchema,
+            })
+          ),
+          pageInfo: z.object({
+            hasNextPage: z.boolean(),
+            endCursor: z.string().nullable(),
+          }),
+        })
+        .nullable(),
+    })
+    .nullable(),
+})
+
+function fetchMyRepos({
+  provider,
+  variables,
+  after,
+  signal,
+}: FetchMyReposArgs) {
+  const query = `
+    query MyRepos($filters: RepositorySetFilters!, $ordering: RepositoryOrdering!, $direction: OrderingDirection!, $after: String) {
+        me {
+          viewableRepositories(filters: $filters, ordering: $ordering, orderingDirection: $direction, first: $first, after: $after) {
```

Comment:
`m`: There seems to be an error between the variables being declared and the ones being used. You seem to be missing `$first: Int` in your query operation.